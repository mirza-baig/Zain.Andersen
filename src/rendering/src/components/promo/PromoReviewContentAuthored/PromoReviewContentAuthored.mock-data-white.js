const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'PromoReviewContentAuthored',
  },
  fields: {
    "additionalInfo": {
      "value": "City, State"
    },
    "reviewerName": {
      "value": "Reviewer Name"
    },
    "reviewStars": {
      "value": 3.4
    },
    "sectionId": {
      "value": ""
    },
    "componentSpacing": null,
    "headlineLevel": {
      "id": "7f10288a-beb1-4d95-a139-32df40094441",
      "url": "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
      "name": "Heading 3",
      "displayName": "Heading 3",
      "fields": {
        "Value": {
          "value": "h3"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "headlineText": {
      "value": "Headline Text"
    },
    "body": {
      "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
    },
    "backgroundColor": {
      "id": "64333070-0d1b-4a33-9e19-7d299d6e8dd3",
      "url": "https://preview.local.andersenwindows.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Background-Color/White",
      "name": "White",
      "displayName": "White",
      "fields": {
        "Value": {
          "value": "white"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "primaryImage": {
      "value": {}
    },
    "primaryImageMobileFocusArea": null,
    "primaryImageMobile": {
      "value": {}
    },
    "primaryVideo": null,
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    }
  },
};

export default defaultData;
