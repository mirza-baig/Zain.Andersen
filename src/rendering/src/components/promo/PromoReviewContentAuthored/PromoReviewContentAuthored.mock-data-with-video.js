const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'PromoReviewContentAuthored',
  },
  fields: {
    "additionalInfo": {
      "value": "City, State"
    },
    "reviewerName": {
      "value": "Reviewer Name"
    },
    "reviewStars": {
      "value": 3.4
    },
    "sectionId": {
      "value": ""
    },
    "componentSpacing": null,
    "headlineLevel": {
      "id": "7f10288a-beb1-4d95-a139-32df40094441",
      "url": "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
      "name": "Heading 3",
      "displayName": "Heading 3",
      "fields": {
        "Value": {
          "value": "h3"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "headlineText": {
      "value": "Headline Text"
    },
    "body": {
      "value": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat."
    },
    "backgroundColor": null,
    "primaryImage": {
      "value": {}
    },
    "primaryImageMobileFocusArea": null,
    "primaryImageMobile": {
      "value": {}
    },
    "primaryVideo": {
      "id": "f6ac1524-c331-4d30-a1f4-716c8edda182",
      "url": "/Data/Elements/Videos/Promo-genric-YouTube-Video169",
      "name": "Promo genric YouTube Video169",
      "displayName": "Promo genric YouTube Video169",
      "fields": {
        "youTubeModestBranding": {
          "value": true
        },
        "youTubeMute": {
          "value": false
        },
        "youTubeShowControls": {
          "value": false
        },
        "youTubeAutoLoop": {
          "value": true
        },
        "youTubeAutoPlay": {
          "value": false
        },
        "youTubeDisableKeyboard": {
          "value": true
        },
        "youTubeClosedCaptions": {
          "value": true
        },
        "videoHeight": {
          "value": 338
        },
        "videoLazyLoad": {
          "value": false
        },
        "videoId": {
          "value": "K-eDe8GDMUo"
        },
        "videoWidth": {
          "value": 600
        }
      },
      "templateId": "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
      "templateName": "YouTube Video"
    },
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    }
  },
};

export default defaultData;
