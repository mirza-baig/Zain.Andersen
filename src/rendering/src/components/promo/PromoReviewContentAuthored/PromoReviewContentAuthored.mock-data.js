const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'PromoReviewContentAuthored',
  },
  fields: {
    "additionalInfo": {
      "value": "City, State"
    },
    "reviewerName": {
      "value": "Reviewer Name"
    },
    "reviewStars": {
      "value": 3.4
    },
    "sectionId": {
      "value": ""
    },
    "componentSpacing": null,
    "headlineLevel": {
      "id": "7f10288a-beb1-4d95-a139-32df40094441",
      "url": "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
      "name": "Heading 3",
      "displayName": "Heading 3",
      "fields": {
        "Value": {
          "value": "h3"
        }
      },
      "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
      "templateName": "Enum"
    },
    "headlineText": {
      "value": "Headline Text"
    },
    "body": {
      "value": 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="~/link.aspx?_id=0AF8C545074B466E93CDB14BF83DB353&amp;_z=z">nulla pariatur</a>. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est<a rel="noopener noreferrer" href="https://loremipsum.io/" target="_blank"> laborum</a> link.<ul><li>bullet point 1</li><li>bullet point 2</li><li>bullet point 3</li></ul>'
    },
    "backgroundColor": null,
    "primaryImage": {
      "value": {}
    },
    "primaryImageMobileFocusArea": null,
    "primaryImageMobile": {
      "value": {}
    },
    "primaryVideo": null,
    "eventName": {
      "value": ""
    },
    "eventType": {
      "value": ""
    },
    "eventZone": {
      "value": ""
    }
  },
};

export default defaultData;
