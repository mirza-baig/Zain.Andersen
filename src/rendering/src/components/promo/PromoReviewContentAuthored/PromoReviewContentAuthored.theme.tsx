// Lib
import classNames from 'classnames';
import { ThemeFile } from 'lib/context/ThemeContext';
import { ComponentBackgroundVariants } from 'src/helpers/Component/Component';

const getAWCopyContainerClass = (hasImage: boolean): string => {
  let copyContainerClass = 'col-span-12 flex flex-col lg:self-center z-[1]';

  if (hasImage) {
    copyContainerClass = classNames(copyContainerClass, 'lg:mr-[20px] lg:col-span-6');
  } else {
    copyContainerClass = classNames(copyContainerClass, 'items-center', 'text-center');
  }

  return copyContainerClass;
};

const getRBACopyContainerClass = (hasImage: boolean): string => {
  let copyContainerClass =
    'col-span-12 lg:flex lg:flex-col lg:self-center z-[1] lg:max-w-screen-ml';

  if (hasImage) {
    copyContainerClass = classNames(copyContainerClass, 'lg:pl-l', 'lg:col-span-7', 'lg:h-full');
  } else {
    copyContainerClass = classNames(copyContainerClass, 'items-start', 'lg:mx-auto');
  }

  return copyContainerClass;
};

const getBaseImageContainerClass = (): string => {
  return classNames('col-span-12', 'lg:content-right', 'lg:items-right');
};

const getAWImageContainerClass = (): string => {
  return classNames(
    getBaseImageContainerClass(),
    'lg:col-span-6',
    'mt-m',
    'lg:mt-0',
    'lg:ml-[20px]'
  );
};

const getRBAImageContainerClass = (): string => {
  return classNames(
    getBaseImageContainerClass(),
    'lg:col-span-5',
    'lg:-order-1',
    'mt-xs',
    'lg:mt-0'
  );
};

const getRBAQuoteClass = (style: ComponentBackgroundVariants): string => {
  const cssClass = 'absolute hidden lg:inline lg:pt-s w-[200px] h-[150px]';

  switch (style) {
    case 'black':
      return classNames(cssClass, 'opacity-[.15] fill-white text-white');
    case 'white':
      return classNames(cssClass, 'opacity-[.05] fill-black text-black');
    case 'gray':
      return classNames(cssClass, 'opacity-[.05] fill-black text-black');
    default:
      break;
  }
  return cssClass;
};

export const PromoReviewContentAuthoredTheme = (
  hasImage: boolean,
  style: ComponentBackgroundVariants
): ThemeFile => {
  return {
    aw: {
      classes: {
        firstHeadline: {
          headlineContainer: `text-theme-text text-sm-m lg:text-m font-medium`,
          headlineOutsideContainer: `${
            hasImage ? '' : 'justify-center'
          } text-theme-text text-sm-m lg:text-m font-medium -mb-s lg:-mb-s mt-8 lg:mx-auto lg:max-w-5xl px-m lg:px-0`,
        },
        contentClasses: {
          componentClass: `p-m ${hasImage ? 'pt-s' : 'pt-m'} lg:pt-m lg:px-0 lg:mx-auto max-w-5xl`,
          starContainerClass: ``,
          body: `text-theme-body text-sm-xxs leading-[1.125rem] lg:text-body lg:leading-[1.125rem] font-regular lg:max-w-3xl mt-m`,
          imageContainerClass: getAWImageContainerClass(),
          copyContainerClass: getAWCopyContainerClass(hasImage),
          reviewerNameClass: `text-theme-text text-sm-xxs leading-[1em] lg:text-xxs lg:leading-[1em] mt-m font-heavy`,
          additionalInfoClass: `text-theme-text text-sm-xxs leading-[1.125rem] lg:text-xxs lg:leading-[1.125rem] mt-2`,
          quoteClass: ``,
        },
        buttonGroupClass: {
          wrapper: classNames('col-span-12', 'flex', 'flex-col', 'lg:flex-row', 'items-start'),
          cta1Classes: classNames(
            'lg:mr-3 lg:first:ml-0 lg:last:mr-0',
            'bg-theme-btn-bg text-theme-btn-text'
          ),
          cta2Classes: '',
        },
      },
    },
    rba: {
      classes: {
        firstHeadline: {
          headlineContainer: `text-theme-text text-s lg:text-m`,
          headlineOutsideContainer: `text-theme-text text-sm-m lg:text-l font-extra-light -mb-s lg:-mb-[10px] mt-8 lg:mx-auto lg:max-w-screen-lg px-m lg:px-0`,
        },
        contentClasses: {
          componentClass: `px-m pt-s pb-m lg:px-0 lg:py-ml relative lg:mx-auto max-w-screen-lg`,
          starContainerClass: `lg:mt-xxs`,
          body: `font-sans text-theme-body text-body lg:text-s font-extra-light mt-[7px] ${
            hasImage ? 'grow lg:mt-[10px]' : 'lg:mt-[9px]'
          }`,
          imageContainerClass: getRBAImageContainerClass(),
          copyContainerClass: getRBACopyContainerClass(hasImage),
          reviewerNameClass: `text-theme-text text-body font-bold h-[18px] ${
            hasImage ? 'mt-[7px]' : 'mt-[7px] lg:mt-[11px]'
          }`,
          additionalInfoClass: `text-theme-text text-body h-[18px] ${
            hasImage ? 'mt-[1px] lg:mt-0' : 'mt-[1px]'
          } lg:mb-[5px]`,
          quoteClass: getRBAQuoteClass(style),
        },
        buttonGroupClass: {
          wrapper: classNames('col-span-12', 'flex', 'flex-col', 'lg:flex-row', 'items-start'),
          cta1Classes: classNames(
            'lg:mr-3 lg:first:ml-0 lg:last:mr-0',
            'bg-theme-btn-bg text-theme-btn-text'
          ),
          cta2Classes: 'my-s md:my-0',
        },
      },
    },
  };
};
