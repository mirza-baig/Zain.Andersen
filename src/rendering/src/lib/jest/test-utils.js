// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
