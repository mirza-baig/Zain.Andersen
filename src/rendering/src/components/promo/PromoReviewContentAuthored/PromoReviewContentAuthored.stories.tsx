// Globalimport { Story, Meta } from '@storybook/react';// Localimport PromoReviewContentAuthored, {  PromoReviewContentAuthoredProps,} from './PromoReviewContentAuthored';import blackData from './PromoReviewContentAuthored.mock-data-black';import defaultData from './PromoReviewContentAuthored.mock-data';import whiteData from './PromoReviewContentAuthored.mock-data-white';import withImage from './PromoReviewContentAuthored.mock-data-with-image';import withVideo from './PromoReviewContentAuthored.mock-data-with-video';export default {  title: 'Promo/PromoReviewContentAuthored',  component: PromoReviewContentAuthored,} as Meta;const Template: Story<PromoReviewContentAuthoredProps> = (props) => (  <PromoReviewContentAuthored {...props} />);export const Default = Template.bind({});Default.args = defaultData;export const Black = Template.bind({});Black.args = blackData;export const White = Template.bind({});White.args = whiteData;export const WithImage = Template.bind({});WithImage.args = withImage;export const WithVideo = Template.bind({});WithVideo.args = withVideo;