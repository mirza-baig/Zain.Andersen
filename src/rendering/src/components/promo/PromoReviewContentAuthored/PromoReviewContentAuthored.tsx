// Global
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { ThemeFile, ThemeName, useTheme } from 'lib/context/ThemeContext';
import { getEnum } from 'lib/utils';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';

// Components
import { Component } from 'src/helpers/Component';
import { ComponentBackgroundVariants } from 'src/helpers/Component/Component';
import { Headline } from 'src/helpers/Headline';
import { MediaPrimary } from 'src/helpers/Media';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import { StarRating, StarRatingProps } from 'src/helpers/StarRating';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { PromoReviewContentAuthoredTheme } from './PromoReviewContentAuthored.theme';

export type PromoReviewContentAuthoredProps =
  Feature.EnterpriseWeb.Enterprise.Components.Promo.PromoReviewContentAuthored.PromoReviewContentAuthored;
const maxStarRating = 5;

const getAWPromoReview = (
  props: PromoReviewContentAuthoredProps,
  themeData: ThemeFile[ThemeName] | undefined,
  hasMedia: boolean
): JSX.Element => {
  const starRatingProps: StarRatingProps = {
    reviewStars: props?.fields?.reviewStars?.value,
    maxStars: maxStarRating,
    containerCSSClass: themeData.classes.contentClasses.starContainerClass,
  };

  return (
    <>
      <div className={themeData.classes.contentClasses.copyContainerClass}>
        <StarRating {...starRatingProps} />
        <RichTextWrapper
          field={props?.fields?.body}
          classes={themeData.classes.contentClasses?.body}
        />
        <div className={themeData.classes.contentClasses.reviewerNameClass}>
          {props?.fields?.reviewerName?.value}
        </div>
        <div className={themeData.classes.contentClasses.additionalInfoClass}>
          {props?.fields?.additionalInfo?.value}
        </div>
      </div>
      {hasMedia && (
        <div className={themeData.classes.contentClasses.imageContainerClass}>
          <MediaPrimary {...props} />
        </div>
      )}
    </>
  );
};

const getRBAPromoReview = (
  props: PromoReviewContentAuthoredProps,
  themeData: ThemeFile[ThemeName] | undefined,
  hasMedia: boolean
): JSX.Element => {
  const starRatingProps: StarRatingProps = {
    reviewStars: props?.fields?.reviewStars?.value,
    maxStars: maxStarRating,
    containerCSSClass: themeData.classes.contentClasses.starContainerClass,
  };

  return (
    <>
      {!hasMedia && (
        <SvgIcon icon="quote" className={themeData.classes.contentClasses.quoteClass} />
      )}
      <div className={themeData.classes.contentClasses.copyContainerClass}>
        <StarRating {...starRatingProps} />
        <RichTextWrapper
          field={props?.fields?.body}
          classes={themeData.classes.contentClasses?.body}
        />
        <div className={themeData.classes.contentClasses.reviewerNameClass}>
          {props?.fields?.reviewerName?.value}
        </div>
        <div className={themeData.classes.contentClasses.additionalInfoClass}>
          {props?.fields?.additionalInfo?.value}
        </div>
      </div>
      {hasMedia && (
        <div className={themeData.classes.contentClasses.imageContainerClass}>
          <MediaPrimary {...props} />
        </div>
      )}
    </>
  );
};

const PromoReviewContentAuthored = (props: PromoReviewContentAuthoredProps) => {
  const hasMedia = !!props.fields?.primaryImage?.value?.src || !!props.fields?.primaryVideo;
  const style = getEnum<ComponentBackgroundVariants>(props.fields?.backgroundColor) ?? 'gray';
  const { themeName, themeData } = useTheme(PromoReviewContentAuthoredTheme(hasMedia, style));

  return (
    <>
      <Headline classes={themeData.classes.firstHeadline.headlineOutsideContainer} {...props} />
      <Component
        variant="full"
        dataComponent="promo/promoreviewcontentauthored"
        gap="gap-x-0"
        padding={themeData.classes.contentClasses.componentClass}
        backgroundVariant={style}
        {...props}
      >
        {themeName === 'aw'
          ? getAWPromoReview(props, themeData, hasMedia)
          : getRBAPromoReview(props, themeData, hasMedia)}
      </Component>
    </>
  );
};

export default withDatasourceCheck()<PromoReviewContentAuthoredProps>(PromoReviewContentAuthored);
