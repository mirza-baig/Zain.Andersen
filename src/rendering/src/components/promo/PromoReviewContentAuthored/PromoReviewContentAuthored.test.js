// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import PromoReviewContentAuthored from './PromoReviewContentAuthored';
import defaultData from './PromoReviewContentAuthored.mock-data';

it('renders correctly', async () => {
  const component = snapshot(PromoReviewContentAuthored, { componentProps: defaultData });
  await hasDataComponent(component, 'promo/promoreviewcontentauthored');
});
