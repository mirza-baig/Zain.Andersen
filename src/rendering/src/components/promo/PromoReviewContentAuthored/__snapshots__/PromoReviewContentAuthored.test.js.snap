// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="items-top font-sans [&_a:hover]:underline justify-center text-theme-text text-sm-m lg:text-m font-medium -mb-s lg:-mb-s mt-8 lg:mx-auto lg:max-w-5xl px-m lg:px-0"
>
  <h3>
    Headline Text
  </h3>
</div>
`;
